posts/2022/batch-processing-notebooks-with-papermill.md

```diff
@@ -0,0 +1,142 @@
+---
+author: Anderson Banihirwe, Matthew Long
+date: 2022-03-14
+tags: python, jupyter, notebooks, papermill, automation
+---
+
+# Batch Processing Jupyter Notebooks with Papermill
+
+## Background and Motivation
+
+[Jupyter notebooks](https://jupyter.org/) are really good for doing the heavy lifting of data analysis by:
+
+- Allowing you to showcase your work in a single place. You can see the complete "paper trail" of what was done. This includes the code, results, visuals and the narrative accompanying your analysis.
+
+- Allowing others to easily use your work as a starting point for their own analysis. A new user can run cell by cell through the notebook to better get an understanding of the code.
+
+However, Jupyter notebooks have a few drawbacks:
+
+- Jupyter notebooks are difficult to maintain and reuse. Unlike a regular Python module that you can import and use in any Python project, users are copying and pasting snippets from each other's notebooks, and it's very easy to get out of sync.
+- Jupyter notebooks are hard to parameterize. This makes it difficult to maintain one version of the truth that can be used as a notebook template for exploring different parameters.
+- Unlike a regular Python scripts that can be run from the command line, running a Jupyter notebook in batch mode requires additional setup and configuration.
+
+Some of these drawbacks can be addressed with the help of Papermill.
+
+## What is Papermill?
+
+[**Papermill**](https://papermill.readthedocs.io/en/latest/) is a Python library that aims to allow users to run and parametrize Jupyter notebooks in a way that is easy to maintain and reuse. In this post, we will walk through how to use Papermill's approach to parametrize a notebook that loads an xarray dataset and plots a map of the data.
+
+## How to use Papermill
+
+### Step 1: Prepare the notebook
+
+To transform your notebook into a Papermill-enabled notebook (a notebook that can be run with Papermill), you need to add the `parameters` tag to cells that contains the parameters you intend to parametrize when running the notebook with Papermill.
+
+![](../../images/papermill-parameters.png)
+
```

Comment:
I think this section needs more explanation.  To a Jupyter newbie, it will not be clear that you need to "click on the gears" icon on the right of the Jupyter page to bring up the "Property Inspector" to change metadata of the cell, which is where all the cell "tags" are located.

Reply:
Done!